### ISSUE TITLE
VideoCapture's index of webcam changes when using rasberry pi

### ISSUE THREAD

**Issue by FayssalProg** [opencv/opencv#23415]
### System Information

OpenCV python version: 4.7.0.68
Operating System: Windows 11
Python version: 3.10

### Detailed description

When running the code to take a videocapture 

```
import cv2
cam = cv2.VideoCapture(0)
```

I wondered if there was a way to always access the webcam of any laptop with the same code, because the index used differs depending on the laptop that runs the code.
For some, its when the index is 0, for others its 1, we even had one where the webcam was the index was 4.

I dont know if it's actually possible but it would be great.

### Steps to reproduce

By having an external camera and trying to access the webcam in different OS.

### Issue submission checklist

- [X] I report the issue, it's not a question
- [X] I checked the problem with documentation, FAQ, open issues, forum.opencv.org, Stack Overflow, etc and have not found any solution
- [X] I updated to the latest OpenCV version and the issue is still there
- [X] There is reproducer code and related data files (videos, images, onnx, etc)

**Comment by saurabhmj11** [comment]
Yes, it's possible for the index of a webcam to change when using a Raspberry Pi. This is because the Raspberry Pi has different USB ports, and each port may be assigned a different index number by the operating system.

When you connect a webcam to a Raspberry Pi, it may be assigned a different index number than when it is connected to a different computer. You can check the index number of the webcam on your Raspberry Pi by running the following command in the terminal:


Copy code
ls /dev/video*
This will list all the video devices that are currently connected to your Raspberry Pi, along with their index numbers.

If you're using OpenCV's VideoCapture function to capture video from the webcam, you'll need to update the index number to match the index number of the webcam on your Raspberry Pi. You can do this by passing the correct index number to the VideoCapture constructor. For example, if your webcam has an index number of 0 on your Raspberry Pi, you can create a VideoCapture object like this:


Copy code
import cv2

cap = cv2.VideoCapture(0)
Note that the index number may change if you disconnect the webcam and connect it to a different USB port on your Raspberry Pi. In that case, you'll need to run the ls /dev/video* command again to determine the new index number

**Comment by MarcelloDM** [comment]
Hi guys, thanks for opening this issue.  
The most simple way to avoid this "random relating" is to create a custom rule for the device. Follow the following steps. 

1. Open a shell session and type check the USB connected list with `lsusb` 

2. Find the device you want to assign a fixed index. Notice  the line contains a fixed couple of **VENDOR_ID:PRODUCT_ID** as such as "**xxxx:yyyy**". In Example (refers to a WebCam) in my case is: 
`Bus 001 Device 008: ID 0c45:6366 Microdia Webcam Vitade AF`

4. Open the Kernel rules file using  `sudo nano /etc/udev/rules.d/99-com.rules`
5. Add a specific rule like this:
`KERNEL=="video0", ATTRS{idVendor}=="0c45", ATTRS{idProduct}=="6366", SYMLINK+="WEBCAM"`

KERNEL is the Device Type (consult Debian documentation for details) and SYMLINK is a arbitrary dummy name for the device, simple to remember. 

Thanks for your work.
